### PR TITLE
Fixes #25748 - fallback to login in user drop down

### PR DESCRIPTION
**Pull request by ares**
To reproduce:

1) login as a user that does not have first name and last name set
2) look at user dropdown where you normally see the name
3) there's nothing

We used to fall back to login in such case, this was dropped when we moved to new implementation in react for layout. We only print first name and last name now. This PR adds `name` virtual attribute that we use in erb helpers to `user.to_json` so it can be easily reused in JS layer. The other option would be to implement this logic in JS, but I'd need some help there + it would duplicate what we already have on server side. @theforeman/ui-ux what do you think, where does this logic belong to? If you prefer JS, where do I put such a helper (it would be useful in other components too)

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -320,6 +320,7 @@ def setup_auditing
     end
 
     def notify_deprecations
+      return true
```
Comment (amirfefer) [inline config/application.rb]: does it related to this PR?

**Review by amirfefer — CHANGES_REQUESTED** [review]
Thanks @ares :+1: ,
Just one question

File: config/application.rb (hunk 1 above)
Comment (ares) [reply config/application.rb]: Oops, no sorry, will remove it asap

**Comment by ares** [comment]
@amirfefer repushed without the return, merry christmas :christmas_tree: :gift: :candle: :-)

**Review by amirfefer — COMMENTED** [review]
Thanks @ares and merry christmas :)

I wonder if the fallback to login needs to be only when there are no first name **and** surname.
for example, when there is no first name, only the surname is shown:
page

**Comment by ares** [comment]
That's how it worked before and works now I believe, see how name/to_label is defined at  https://github.com/theforeman/foreman/blob/3900aa94e6952acdca49f0ece4631cf06ccff3ea/app/models/user.rb#L184

File: app/models/user.rb [hunk 2]
```diff
@@ -129,6 +129,10 @@ def self.name_format
 
   attr_exportable :firstname, :lastname, :mail, :description, :fullname, :name => ->(user) { user.login }, :ssh_authorized_keys => ->(user) { user.ssh_keys.map(&:to_export_hash) }
 
+  def as_json(options = {})
+    super.as_json(options).tap { |u| u['user']['name'] = name }
```
Comment (tbrisker) [inline app/models/user.rb]: doesn't `super` already call `as_json(options)` on the parent class?

Comment (ares) [reply app/models/user.rb]: yup, updated, calling just `super`

**Review by tbrisker — DISMISSED** [review]
APJ, Thanks @ares !

**Comment by tbrisker** [comment]
@ares travis failure seems related, please check.

**Comment by tbrisker** [comment]
and foreman tests as well...

**Comment by ares** [comment]
both should be resolved now

File: webpack/assets/javascripts/react_app/components/Layout/components/__snapshots__/UserDropdowns.test.js.snap [hunk 3]
```diff
@@ -38,9 +38,6 @@ exports[`UserDropdown rendering render switcher w/loading 1`] = `
         name="user avatar small"
         type="fa"
       />
-      G
-       
-      L
```
Comment (tbrisker) [inline webpack/assets/javascripts/react_app/components/Layout/components/__snapshots__/UserDropdowns.test.js.snap:43]: I think this needs fixing in https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/components/Layout/Layout.fixtures.js#L192 to match the changes server response

**Comment by tbrisker** [comment]
@ares travis failure related

**Review by tbrisker — APPROVED** [review]
Thanks @ares !

**Pull request merged**